test/core/handshake/http_proxy_mapper_test.cc

```diff
@@ -224,6 +224,30 @@ TEST(ProxyForAddressTest, BadProxy) {
   EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER), std::nullopt);
 }
 
+TEST(ProxyForAddressTest, UserInfo) {
+  auto args = ChannelArgs().Set(GRPC_ARG_HTTP_PROXY,
+                                "http://username:password@proxy.google.com");
+  EXPECT_EQ(HttpProxyMapper().MapName("dns:///test.google.com:443", &args),
+            "proxy.google.com");
+  EXPECT_EQ(args.GetString(GRPC_ARG_HTTP_CONNECT_SERVER),
+            "test.google.com:443");
+  EXPECT_EQ(
+      args.GetString(GRPC_ARG_HTTP_CONNECT_HEADERS),
+      absl::StrCat("Proxy-Authorization:Basic ", "dXNlcm5hbWU6cGFzc3dvcmQ="));
```

Comment:
No need for `absl::StrCat()` here, since these are both static strings.  We can just combine them into one string.